smarts/core/trap_manager.py

```diff
@@ -265,28 +274,50 @@ def largest_vehicle_plane_dimension(vehicle: Vehicle):
                     if overlapping:
                         continue
 
-                vehicle = TrapManager._make_vehicle(
-                    sim, agent_id, mission, trap.default_entry_speed
+                vehicle = TrapManager._make_new_vehicle(
+                    sim,
+                    agent_id,
+                    mission,
+                    trap.default_entry_speed,
+                    social=agent_id in sim.agent_manager.pending_social_agent_ids,
                 )
             else:
                 continue
-            if vehicle == None:
+            if vehicle is None:
                 continue
-            sim.create_vehicle_in_providers(vehicle, agent_id, True)
-            agents_given_vehicle.add(agent_id)
             used_traps.append((agent_id, trap))
 
-        if len(agents_given_vehicle) > 0:
+        if len(used_traps) > 0:
             self.remove_traps(used_traps)
-            sim.agent_manager.remove_pending_agent_ids(agents_given_vehicle)
 
     @property
     def traps(self) -> Dict[str, Trap]:
         """The traps in this manager."""
         return self._traps
 
     @staticmethod
-    def _make_vehicle(sim, agent_id, mission, initial_speed):
+    def _take_existing_vehicle(sim, vehicle_id, agent_id, mission, social=False):
+        from smarts.core.smarts import SMARTS
+
+        assert isinstance(sim, SMARTS)
+        if social:
+            # Not supported
+            return None
```

Comment:
Unsure if I should support the social agent taking over social vehicles, it may be reasonable but only if I add another entry tactic.